cpp_exercise/src/lib/linuxipclib.h

```diff
@@ -0,0 +1,109 @@
+#ifndef LINUXIPCLIB_H
+#define LINUXIPCLIB_H
+
+#include <map>
+#include <string>
+#include <vector>
+#include <iostream>
+#include <map>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <cstring>
+#include <sys/types.h>
+#include <vector>
+#include <getopt.h>
+#include "ipcpipelib.h"
+
+
+namespace ipcprt {
+
+    enum Protocol {
```

Comment:
If you will use 'enum class' maybe you won't need this namespace ?

Reply:
I actually used the namespace becuz there I used "pipe" in this enum and there is also the declaration in unistd.h library which gives me redeclaration error If I don't use namespace